src/writers/writer_postgres.py

```
@@ -276,51 +182,49 @@ def write(self, topic_name: str, message: dict[str, Any]) -> tuple[bool, str | N
-                    if topic_name == "public.cps.za.dlchange":
-                        self._postgres_edla_write(cursor, table_info["main"], message)
-                    elif topic_name == "public.cps.za.runs":
-                        self._postgres_run_write(cursor, table_info["main"], table_info["jobs"], message)
-                    elif topic_name == "public.cps.za.test":
-                        self._postgres_test_write(cursor, table_info["main"], message)
-
-                connection.commit()
+            self._execute_with_retry(lambda conn: self._write_topic(conn, topic_name, message))
         except (RuntimeError, PsycopgError, BotoCoreError, ClientError, ValueError, KeyError) as e:
+            self._close_connection()
             err_msg = f"The Postgres writer failed with unknown error: {str(e)}"
```

Ruff (0.15.10)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)